### PR TITLE
[v6r15] Enable CodeDocumentation via READTHEDOCS

### PR DESCRIPTION
**Pull request by andresailer**
In the Read the Docs configuration for dirac
under "Admin" and "Advanced settings put "docs/source/conf.py" into "Python Configuration File"
and "docs/requirements.txt" into "Requirements file".

See the result here:
http://diracsailer.readthedocs.io/en/rel-v6r15/CodeDocumentation/index.html
To run this locally:
```
cd DIRAC/docs; READTHEDOCS=True make html
```

I added a MakeDoc script (couldn't figure out the buildCodeDoc). This script creates documentation stumps for all dirac code (except for PlotCache and PlottingHandler and some scripts not in scripts folders, because those crash or hang the sphinx build).

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -11,37 +11,92 @@
...
-  diracRelease = os.environ[ 'DIRACVERSION' ]
-except KeyError:
-  diracRelease = 'integration'
+  import fakeEnvironment
+except ImportError:
+  pass
+try:
+  import fakeEnv
+except ImportError:
+  pass
```
Comment (fstagni) [inline docs/source/conf.py]: Out of curiosity, what's the goal of the lines above?

Comment (fstagni) [inline docs/source/conf.py]: Ah IIUC is mocking GSI

**Comment by fstagni** [comment]
BTW: thanks for this.

Review OK.

File: docs/source/conf.py (hunk 1 above)
Comment (andresailer) [inline docs/source/conf.py]: Yes, exactly. This is mocking all external packages. the fakeEnvironment.py was copied from the docs/Tools folder and updated for all ImportErrors during the documentation creation.

**Pull request merged**